packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts

```diff
@@ -126,56 +172,49 @@ export const listShippingOptionsForCartWorkflow = createWorkflow(
 
         "calculated_price.*",
       ],
-      variables: {
-        context: {
-          is_return: !!input.is_return,
-          enabled_in_store: "true",
-        },
-        filters: {
-          fulfillment_set_id: fulfillmentSetIds,
-          address: {
-            city: cart.shipping_address?.city,
-            country_code: cart.shipping_address?.country_code,
-            province_code: cart.shipping_address?.province,
-          },
-        },
-
-        calculated_price: {
-          context: pricingContext,
-        },
-      },
+      variables: queryVariables,
     }).config({ name: "shipping-options-query" })
 
-    const shippingOptionsWithPrice = transform({ shippingOptions }, (data) => {
-      const optionsMissingPrices: string[] = []
+    const updatedShippingOptions = transform(
+      { shippingOptions, input },
+      ({ shippingOptions, input }) => {
+        const optionsMissingPrices: string[] = []
 
-      const options = data.shippingOptions.map((shippingOption) => {
-        const { calculated_price, ...options } = shippingOption ?? {}
+        const options = shippingOptions.map((shippingOption) => {
+          const { calculated_price, ...options } = shippingOption
 
-        if (options?.id && !isPresent(calculated_price?.calculated_amount)) {
-          optionsMissingPrices.push(options.id)
-        }
+          if (
+            shippingOption?.id &&
+            !isDefined(calculated_price?.calculated_amount)
+          ) {
+            optionsMissingPrices.push(options.id)
+          }
 
-        return {
-          ...options,
-          amount: calculated_price?.calculated_amount,
-          is_tax_inclusive:
-            !!calculated_price?.is_calculated_price_tax_inclusive,
-        }
-      })
+          return {
+            ...shippingOption,
+            amount: calculated_price?.calculated_amount,
+            is_tax_inclusive:
+              !!calculated_price?.is_calculated_price_tax_inclusive,
+          }
+        })
+
+        const shouldValidatePrices = input.options?.shouldValidatePrices ?? true
```

Comment:
thought: What's the idea with this option? When do we not want to validate prices

Reply:
When we create prices for shipping options,we validate them. 

When we're updating price during a cart op, we remove the ones with invalid prices

Happy to choose another way for this, didn't like it either.

Reply:
I'd almost rather have two steps – one for validating and one without it. I think it's not obvious what's going on here. Wdyt?